src/wallet/rpcwallet.cpp

```diff
@@ -3261,8 +3264,8 @@ static UniValue bumpfee(const JSONRPCRequest& request)
                 },
                 RPCResult{
                     RPCResult::Type::OBJ, "", "", {
-                        {RPCResult::Type::STR, "psbt", "The base64-encoded unsigned PSBT of the new transaction. Only returned when wallet private keys are disabled."},
-                        {RPCResult::Type::STR_HEX, "txid", "The id of the new transaction. Only returned when wallet private keys are enabled."},
+                        {RPCResult::Type::STR, "psbt", "The base64-encoded unsigned PSBT of the new transaction." + std::string(want_psbt ? "" : " Only returned when wallet private keys are disabled. (DEPRECATED)")},
+                        {(want_psbt ? RPCResult::Type::HIDDEN : RPCResult::Type::STR_HEX), "txid", "The id of the new transaction. Only returned when wallet private keys are enabled."},
```

Comment:
I think it doesn't make sense to push an rpc result into the help manager and then tell the manager to discard it. Wouldn't it make more sense to not push it in the first place? This should be trivial to achieve with Cat

```cpp
RPCResult::Type::OBJ, "", "", Cat(Cat({
    {STR, "psbt", ...},
  },
  std::vector<RPCResult>{want_psbt ? {} : {STR_HEX, "txid", ...}}),
  {
    {STR_AMOUNT, "origfee" ...},
    {STR_AMOUNT, "fee", ...},
    ....
  }),
```

(I didnt' compile this to check)

Reply:
I agree that it doesn't make sense to push and then hide. I was originally trying to do something similar to this but couldn't get it to work. I was also unable to get your suggestion to work. It fails with

```
wallet/rpcwallet.cpp:3270:58: error: no matching function for call to ‘std::vector<RPCResult>::vector(<brace-enclosed initializer list>)’
 3270 |                         std::vector<RPCResult>{want_psbt ? {} : {RPCResult::Type::STR_HEX, "txid", "The id of the new transaction. Only returned when wallet private keys are enabled."}
```

I tried a few different variations of this but nothing would work.

Reply:
Ah, not your fault. That `{}` means both "constructor" and "initializer list" is one of the countless design flaws of c++. `{{}}` should work.

This diff compiles (haven't tested what it doesn, though):

```diff
diff --git a/src/wallet/rpcwallet.cpp b/src/wallet/rpcwallet.cpp
index de21f23da3..e5ecc6a451 100644
--- a/src/wallet/rpcwallet.cpp
+++ b/src/wallet/rpcwallet.cpp
@@ -3263,16 +3263,16 @@ static UniValue bumpfee(const JSONRPCRequest& request)
                         "options"},
                 },
                 RPCResult{
-                    RPCResult::Type::OBJ, "", "", {
+                    RPCResult::Type::OBJ, "", "", Cat(Cat({
                         {RPCResult::Type::STR, "psbt", "The base64-encoded unsigned PSBT of the new transaction." + std::string(want_psbt ? "" : " Only returned when wallet private keys are disabled. (DEPRECATED)")},
-                        {(want_psbt ? RPCResult::Type::HIDDEN : RPCResult::Type::STR_HEX), "txid", "The id of the new transaction. Only returned when wallet private keys are enabled."},
+                        },want_psbt ? std::vector<RPCResult>{} : std::vector<RPCResult>{{RPCResult::Type::STR_HEX, "txid", "The id of the new transaction. Only returned when wallet private keys are enabled."}}),{
                         {RPCResult::Type::STR_AMOUNT, "origfee", "The fee of the replaced transaction."},
                         {RPCResult::Type::STR_AMOUNT, "fee", "The fee of the new transaction."},
                         {RPCResult::Type::ARR, "errors", "Errors encountered during processing (may be empty).",
                         {
                             {RPCResult::Type::STR, "", ""},
                         }},
-                    }
+                    })
                 },
                 RPCExamples{
             "\nBump the fee, get the new transaction\'s" + std::string(want_psbt ? "psbt" : "txid") + "\n" +

Reply:
Done. Dropped the commit adding `HIDDEN`